typegate/src/engine.ts

```diff
@@ -219,11 +235,14 @@ export class Engine {
     verbose: boolean,
   ): Promise<JSONValue> {
     const ret = {};
-    const cache: Record<string, unknown> = {};
-    const lenses: Record<string, unknown> = {};
+    const cache: Record<string, unknown[]> = {};
```

Comment:
why is an array required? maybe you can add comment on the example data cached and put in the lenses?